ray-operator/controllers/ray/common/pod.go

```diff
@@ -661,14 +660,6 @@ func setContainerEnvVars(pod *corev1.Pod, rayNodeType rayv1.RayNodeType, gcsOpti
 		}
 		container.Env = append(container.Env, extraTagsEnv)
 	}
-	if !utils.EnvVarExists(utils.REDIS_PASSWORD, container.Env) {
```

Comment:
It's quite odd to set an environment variable to `rayStartParams[redis-password]`. In most cases, users should not explicitly set the `redis-password` in `rayStartParams`. Instead, users should use `$REDIS_PASSWORD` in `rayStartParams` to reference an environment variable that points to a Kubernetes secret.

<img width="1425" alt="image" src="https://github.com/user-attachments/assets/31a7816a-cad4-44bf-95ef-6474543d0847" />

Reply:
It's a breaking change, but I think it is fine:

1. We don't document the behavior where the `REDIS_PASSWORD` environment variable is set to `rayStartParams[redis-password]` if it is not explicitly defined.
2. The side effect is that Redis cleanup may fail in certain special cases. However, since Redis cleanup is a best-effort behavior, this will only result in a Redis memory leak and will not cause Ray jobs to fail.
3. Having two different logics to set the `REDIS_PASSWORD` environment variable adds unnecessary complexity.

Reply:
second thought: let me try to revisit whether we should keep the compatibility or not if we can avoid the complexity